src/main.cpp

```diff
@@ -3852,7 +3852,7 @@ DEMUX_PACKET* CInputStreamAdaptive::DemuxRead(void)
     const AP4_UI08* pData(sr->GetSampleData());
     DEMUX_PACKET* p;
 
-    if (iSize && pData && sr->IsEncrypted())
+    if (sr->IsEncrypted() && iSize > 0 && pData)
```

Comment:
no change is needed here I think - `iSize > 0` check is redundant as it's the same as checking `iSize`

Reply:
numerical variables should be compared in this way,
i was specifically requested to use this rule in the kodi source code
there are different rules between components?

Reply:
No that's fine, I'm not up to speed yet on all the style guidelines. Disregard my comment :)